retrieval/target_test.go

```diff
@@ -283,6 +283,66 @@ func TestNewHTTPClientCert(t *testing.T) {
 	}
 }
 
+func TestNewHTTPWithServerName(t *testing.T) {
+	server := httptest.NewUnstartedServer(
+		http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", `text/plain; version=0.0.4`)
+				w.Write([]byte{})
+			},
+		),
+	)
+	server.TLS = newTLSConfig(t)
+	server.StartTLS()
+	defer server.Close()
+
+	cfg := &config.ScrapeConfig{
+		ScrapeTimeout: model.Duration(1 * time.Second),
+		TLSConfig: config.TLSConfig{
+			CAFile:     "testdata/ca.cer",
+			ServerName: "127.0.0.1",
```

Comment:
This is the same as the address, so this isn't really testing it. Can you add a 2nd cert with a different name?